### PR TITLE
common/blkdev: add missing stubs for OSes not Linux

### PR DESCRIPTION
**Pull request by wjwithagen**
Otherwise the linker wil complain:

'''
usr/local/bin/x86_64-unknown-freebsd14.0-ld: /usr/local/bin/x86_64-unknown-freebsd14.0-ld: /usr/local/bin/x86_64-unknown-freebsd
14.0-ld: ../../lib/libblk.a(KernelDevice.cc.o): in function `KernelDevice::open(std::__1::basic_string<char, std::__1::char_trai
ts<char>, std::__1::allocator<char> > const&)':
/home/jenkins/workspace/ceph-master-compile/src/blk/kernel/KernelDevice.cc:236: undefined reference to `BlkDev::get_optimal_io_s
ize() const'
'''

fixes: #43691
tracker: #53483

## Checklist
- Tracker (select at least one)
  - [x] References tracker ticket
  - [x] Very recent bug; references commit where it was introduced
  - [ ] New feature (ticket optional)
  - [ ] Doc update (no ticket needed)
  - [ ] Code cleanup (no ticket needed)
- Component impact
  - [ ] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [ ] No impact that needs to be tracked
- Documentation (select at least one)
  - [ ] Updates relevant documentation
  - [ ] No doc update is appropriate
- Tests (select at least one)
  - [ ] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [ ] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`

</details>

### CODE REVIEW

**Review by ifed01 — APPROVED** [review]

File: src/common/blkdev.cc [hunk 1]
```diff
@@ -864,6 +864,11 @@ int BlkDev::discard(int64_t offset, int64_t len) const
   return -EOPNOTSUPP;
 }
 
+int BlkDev::get_optimal_io_size() const
+{
+	return 0;
```
Comment (tchaikov) [inline src/common/blkdev.cc:869]: might want to use two spaces instead of a <tab> for indent here.

File: src/common/blkdev.cc [hunk 2]
```diff
@@ -993,6 +998,11 @@ int BlkDev::discard(int64_t offset, int64_t len) const
...
```
Comment (tchaikov) [inline src/common/blkdev.cc:1003]: ditto.

**Review by tchaikov — COMMENTED** [review]
and might want to revise the commit message. currently, it reads

> tracker: #53483
> Signed-off-by: Willem Jan Withagen <wjw@digiware.nl>
> 
> Signed-off-by: Willem Jan Withagen <wjw@digiware.nl>

would be better to

- change `tracker: #53483` to `Fixes: https://tracker.ceph.com/issues/53483`
- dedup the `Signed-off-by:` line

**Comment by wjwithagen** [comment]
@tchaikov 
Fixed

**Review by tchaikov — APPROVED** [review]

**Comment by tchaikov** [comment]
jenkins test api

**Comment by wjwithagen** [comment]
@tchaikov 
Can this be committed?

**Pull request merged**